src/DataStructures/ComplexDataVector.hpp

```diff
@@ -0,0 +1,182 @@
+
+template <typename Operator>
+struct BinaryMapTrait<DataVector, ComplexDataVector, Operator> {
+  using Type = ComplexDataVector;
+};
+template <typename Operator>
+struct BinaryMapTrait<ComplexDataVector, DataVector, Operator> {
+  using Type = ComplexDataVector;
+};
+}  // namespace blaze
+
+DECLARE_VECTOR_EQUIV(ComplexDataVector)
+bool operator==(const ComplexDataVector& lhs, const DataVector& rhs) noexcept;
+bool operator==(const DataVector& lhs, const ComplexDataVector& rhs) noexcept;
+bool operator!=(const ComplexDataVector& lhs, const DataVector& rhs) noexcept;
+bool operator!=(const DataVector& lhs, const ComplexDataVector& rhs) noexcept;
+
+MAKE_ARRAY_VECTOR_BINOPS(ComplexDataVector)
+
+/// \cond HIDDEN_SYMBOLS
+DEFINE_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, DataVector, operator+,
+                   std::plus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, DataVector, ComplexDataVector, operator+,
+                   std::plus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, double, operator+,
+                   std::plus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, double, ComplexDataVector, operator+,
+                   std::plus<>())
+
+DEFINE_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, DataVector, operator-,
+                   std::minus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, DataVector, ComplexDataVector, operator-,
+                   std::minus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, double, operator-,
+                   std::minus<>())
+DEFINE_ARRAY_BINOP(ComplexDataVector, double, ComplexDataVector, operator-,
+                   std::minus<>())
+
+DEFINE_ARRAY_INPLACE_BINOP(ComplexDataVector, DataVector, operator+=,
+                           std::plus<>())
+DEFINE_ARRAY_INPLACE_BINOP(ComplexDataVector, DataVector, operator-=,
+                           std::minus<>())
+/// \endcond
+
+MAKE_VECTOR_MAKEWITHVALUES(ComplexDataVector);
+
+namespace ConstantExpressions_detail {
+template <>
+struct pow<ComplexDataVector, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const ComplexDataVector& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+template <>
+struct pow<std::reference_wrapper<ComplexDataVector>, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const std::reference_wrapper<ComplexDataVector>& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+template <>
+struct pow<std::reference_wrapper<const ComplexDataVector>, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const std::reference_wrapper<const ComplexDataVector>& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+
+template <int N>
+struct pow<ComplexDataVector, N, Requires<(N < 0)>> {
+  static_assert(N > 0,
+                "Cannot use pow on DataVector structures with a negative "
```

Comment:
`DataVector` -> `ComplexDataVector` and I guess the formatting of the string (the awkward line break on the second line) could be fixed :)